flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/CEPITCase.java

```diff
@@ -383,6 +386,100 @@ public String select(Map<String, List<Event>> pattern) {
 		env.execute();
 	}
 
+	@Test
+	public void testSimpleKeyedPatternEventTimeWithSideOutput() throws Exception {
```

Comment:
How about to simplify the test a bit? 

I would be in favour of implementing this test with `OperatorTestHarness` (in CepOperatorTest). This way we it is way easier to follow which events are late.

Also I would use a simpler Pattern here... Don't see a reason for using 3 part one.

Reply:
move the test case to CepOperatorTest, please help review again, thanks ;)